go/internal/proximity-transport/example_test.go

```diff
@@ -0,0 +1 @@
+package proximitytransport_test
```

Comment:
since you moved a lot of logic in this package

it now worths having unit tests against this (future shared) logic